samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -52,6 +52,26 @@ void registerNextOperator(OperatorImpl<RM, ?> nextOperator) {
   public abstract void onNext(M message, MessageCollector collector, TaskCoordinator coordinator);
 
   /**
+   * Invoked at every tick. This method delegates to {@link #onTimer(MessageCollector, TaskCoordinator)}
+   *
+   * @param collector
+   * @param coordinator
+   */
+  public final void onTick(MessageCollector collector, TaskCoordinator coordinator) {
+    onTimer(collector, coordinator);
+    propagateTimer(collector, coordinator);
+  }
+
+  /**
+   * Invoked at every tick. Implementations may call {@link #propagateResult} if they wish to propagate the output to registered
+   * downstream operators.
+   *
+   * @param collector
+   * @param coordinator
+   */
+  public abstract void onTimer(MessageCollector collector, TaskCoordinator coordinator);
```

Comment:
A nice property is that it forces the implementor to think if there's anything to be done periodically in the `timer` callback. (and explicitly define a no-op). We seem to have equal # of stateless and stateful operators, so, it's no big deal either way.

I've fixed as per your suggestion.